chef/automate
# Desktop UI Rough Insight Control

### :nut_and_bolt: Description: What code changed, and why?
Adding the desktop insight control. This control displays a filtered list of desktops. The control is initially hidden and appears when a row is selected from the "Top Errors" control.  The desktops can be sorted by clicking one of the three column headers. This can be sorted in ascending and descending order.  Pagination is also added. There is also a button on the top right to re-hide the control. 

Go to the below link to see how this page should work in the end.
https://chef.invisionapp.com/share/E9W3TEFXV87#/screens/410934037. 
On that page click on the first row of the "Top Errors" control. 
![Screen Shot 2020-04-08 at 1 42 30 PM](https://user-images.githubusercontent.com/1679247/78831643-d4c9a580-799e-11ea-9520-18847261b426.png)


I had a bit of fun creating this. I hope this helps out the UI team in implementing the real control. 

### :chains: Related Resources

#3219

### :athletic_shoe: How to Build and Test the Change
1. start_all_services
1. Run the below command to get some failed desktop runs with differing error types and messages. 
```
for _ in {1..100} ; do
  m=$((RANDOM % 4))
  n=$((RANDOM % 4))
  generate_chef_run_failure_example | \
    jq --arg msg "Error $n occurred" --arg type "Chef::ExampleError$m" '.error.message = $msg | .error.class = $type' | \
    send_chef_data_raw
done
```
1. Start the local automate-ui code in your preferred way. https://github.com/chef/automate/blob/master/dev-docs/ui-development.md
1. Open the desktop page. https://github.com/chef/automate/blob/master/dev-docs/ui-development.md
1. Click one of the rows in the "Top 10 Errors" control. 
1. Ensure the "Insight" control appears and works correctly. 

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable

![insight](https://user-images.githubusercontent.com/1679247/78828838-2facce00-799a-11ea-937d-d26abf7c836e.gif)
